osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs

```diff
@@ -147,17 +156,24 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         /// </summary>
         public void SetTarget(OsuScreen screen)
         {
-            lastTargetScreen = screen;
+            try
+            {
+                lastTargetScreen = screen;
 
-            if (skinEditor == null) return;
+                if (skinEditor == null) return;
 
-            skinEditor.Save(userTriggered: false);
+                skinEditor.Save(userTriggered: false);
 
-            // ensure the toolbar is re-hidden even if a new screen decides to try and show it.
-            updateComponentVisibility();
+                // ensure the toolbar is re-hidden even if a new screen decides to try and show it.
+                updateComponentVisibility();
 
-            // AddOnce with parameter will ensure the newest target is loaded if there is any overlap.
-            Scheduler.AddOnce(setTarget, screen);
+                // AddOnce with parameter will ensure the newest target is loaded if there is any overlap.
+                Scheduler.AddOnce(setTarget, screen);
+            }
+            finally
+            {
+                globallyReenableBeatmapSkinSetting();
+            }
```

Comment:
So `setTarget()` - which internally contains a `globallyDisableBeatmapSkinSetting()` call if the target screen is a `Player` - is scheduled, but then `globallyReenableBeatmapSkinSetting()` is called in a `finally` block? It's probably not *wrong*, as the schedule saves this, but why is this written like that in the first place? I assume the `finally` is aimed at the early-return, but if so, then can the re-enable not be done at the start of the method or something?